src/widget/wglwidgetqopengl.cpp

```diff
@@ -0,0 +1,96 @@
+#include <QMouseEvent>
+#include <QResizeEvent>
+
+#include "widget/openglwindow.h"
+#include "widget/tooltipqopengl.h"
+#include "widget/wglwidget.h"
+
+WGLWidget::WGLWidget(QWidget* parent)
+        : QWidget(parent) {
+}
+
+WGLWidget::~WGLWidget() {
+    ToolTipQOpenGL::singleton().stop(this);
+    if (m_pOpenGLWindow) {
+        m_pOpenGLWindow->widgetDestroyed();
+    }
+}
+
+QPaintDevice* WGLWidget::paintDevice() {
```

Comment:
Unsure here, the main issue is that this is not only a getter like the function name implies.  
Maybe you have an idea for improvement? 

The function describes the usage not what it does.  
Maybe prepareWindowForPainting() or something. 
Since this is also no abstract function we can return the OpenGLWindow*

Reply:
I prefer to keep the name, but to add another function, clearPaintDevice(), and call this explicitly where relevant. Would that work for you?

IIRC this was a stop-gap solution to solve the legacy renderers showing green background on certain hardware.

Reply:
Yes. I have no strong opinion here. I have been just stumbled at this point and wrote that comment for thinking.

Reply:
IIRC this was to fix the issue with the legacy renderers drawing with a green background on certain hardware.

I prefer to leave the function called paintDevice() (and I prefer to keep the return type QPaintDevice* for clarify) and add an explicit additional clearPaintDevice() call where needed.